### ISSUE TITLE
Outer loop Win7 and Win10 builds failing consistently with various errors

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#6493]
Outer loop builds are repeatedly failing with a variety of errors:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/outerloop_win7_debug/98/consoleText
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/outerloop_win10_debug/104/consoleText
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/outerloop_win7_release/94/consoleText
```
09:17:15 Installing dotnet cli...
09:17:16 Add-Type : Cannot add type. The assembly 'System.IO.Compression.FileSystem' cou
09:17:16 ld not be found.
09:17:16 At line:1 char:250
09:17:16 + (New-Object Net.WebClient).DownloadFile('https://dotnetcli.blob.core.windows.
09:17:16 net/dotnet/beta/Binaries/1.0.0.001504/dotnet-win-x64.1.0.0.001504.zip', 'd:\j\w
09:17:16 orkspace\outerloop_win---9c9e7d59\Tools\dotnetcli\dotnet-win-x64.1.0.0.001504.z
09:17:16 ip'); Add-Type <<<<  -Assembly 'System.IO.Compression.FileSystem' -ErrorVariabl
09:17:16 e AddTypeErrors; if ($AddTypeErrors.Count -eq 0) { [System.IO.Compression.ZipFi
09:17:16 le]::ExtractToDirectory('d:\j\workspace\outerloop_win---9c9e7d59\Tools\dotnetcl
09:17:16 i\dotnet-win-x64.1.0.0.001504.zip', 'd:\j\workspace\outerloop_win---9c9e7d59\To
09:17:16 ols\dotnetcli\') } else { (New-Object -com shell.application).namespace('d:\j\w
09:17:16 orkspace\outerloop_win---9c9e7d59\Tools\dotnetcli\').CopyHere((new-object -com 
09:17:16 shell.application).namespace('d:\j\workspace\outerloop_win---9c9e7d59\Tools\dot
09:17:16 netcli\dotnet-win-x64.1.0.0.001504.zip').Items(),16) }
09:17:16     + CategoryInfo          : ObjectNotFound: (System.IO.Compression.FileSyste 
09:17:16    m:String) [Add-Type], Exception
09:17:16     + FullyQualifiedErrorId : ASSEMBLY_NOT_FOUND,Microsoft.PowerShell.Commands 
09:17:16    .AddTypeCommand
09:17:16  
09:17:17 Add-Type : Cannot add type. One or more required assemblies are missing.
09:17:17 At line:1 char:250
09:17:17 + (New-Object Net.WebClient).DownloadFile('https://dotnetcli.blob.core.windows.
09:17:17 net/dotnet/beta/Binaries/1.0.0.001504/dotnet-win-x64.1.0.0.001504.zip', 'd:\j\w
09:17:17 orkspace\outerloop_win---9c9e7d59\Tools\dotnetcli\dotnet-win-x64.1.0.0.001504.z
09:17:17 ip'); Add-Type <<<<  -Assembly 'System.IO.Compression.FileSystem' -ErrorVariabl
09:17:17 e AddTypeErrors; if ($AddTypeErrors.Count -eq 0) { [System.IO.Compression.ZipFi
09:17:17 le]::ExtractToDirectory('d:\j\workspace\outerloop_win---9c9e7d59\Tools\dotnetcl
09:17:17 i\dotnet-win-x64.1.0.0.001504.zip', 'd:\j\workspace\outerloop_win---9c9e7d59\To
09:17:17 ols\dotnetcli\') } else { (New-Object -com shell.application).namespace('d:\j\w
09:17:17 orkspace\outerloop_win---9c9e7d59\Tools\dotnetcli\').CopyHere((new-object -com 
09:17:17 shell.application).namespace('d:\j\workspace\outerloop_win---9c9e7d59\Tools\dot
09:17:17 netcli\dotnet-win-x64.1.0.0.001504.zip').Items(),16) }
09:17:17     + CategoryInfo          : InvalidData: (:) [Add-Type], InvalidOperationExc 
09:17:17    eption
09:17:17     + FullyQualifiedErrorId : ASSEMBLY_LOAD_ERRORS,Microsoft.PowerShell.Comman 
09:17:17    ds.AddTypeCommand
09:17:17  
09:17:19 Restoring BuildTools version 1.0.25-prerelease-00177...
09:17:24 Initializing BuildTools ...
09:17:37 Done initializing tools.
09:17:37 [17:17:45.06] Building Native Libraries...
09:17:38 CMake is a pre-requisite to build this repository but it was not found on the path. Please install CMake from http://www.cmake.org/download/ and ensure it is on your path.
09:17:38 Native component build failed see nativebuild.log for more details.
09:17:38 [17:17:45.81] Building Managed Libraries...
09:17:39   [17:17:46.59] Restoring all packages...
09:21:19   [17:21:26.71] Restoring all packages...Done.
09:21:20 C:\Program Files (x86)\MSBuild\Microsoft\Portable\Microsoft.Portable.Core.targets(22,9): error : The TargetFrameworkProfile property is not set for project 'Microsoft.CSharp.csproj'. Portable projects must specify a profile. [d:\j\workspace\outerloop_win---9c9e7d59\src\Microsoft.CSharp\ref\Microsoft.CSharp.csproj]
09:21:21 C:\Program Files (x86)\MSBuild\Microsoft\Portable\Microsoft.Portable.Core.targets(22,9): error : The TargetFrameworkProfile property is not set for project 'Microsoft.VisualBasic.csproj'. Portable projects must specify a profile. [d:\j\workspace\outerloop_win---9c9e7d59\src\Microsoft.VisualBasic\ref\Microsoft.VisualBasic.csproj]
```

**Comment by Priya91** [comment]
The cmake and the System.IO.Compression.FileSystem not found errors have been fixed. Those were due to incorrect win7 images. But then the build is failing due to the TargetFrameworkProfile property not set errors.. 

Regarding the msbuild target errors, @mellinoe has worked on this in the [commit](https://github.com/dotnet/corefx/commit/41433cafa28dd0759c7b42cf41c9c6f019b5fe4e) .. Can you take a look..

**Comment by akoeplinger** [comment]
The `assembly 'System.IO.Compression.FileSystem' could not be found.` error popped up on gitter yesterday too, it's because Win7 ships with PowerShell 2 out of the box which doesn't have that. The script should check for min PS version.

/cc @masonwheeler

**Comment by masonwheeler** [comment]
Thanks @akoeplinger.  Yeah, this is a problem on coreclr too, for the same reason.  Reported at https://github.com/dotnet/coreclr/issues/3510

**Comment by Priya91** [comment]
@akoeplinger I fixed this by upgrading the powershell version on the win7 azure images.. This should unblock the builds on win7.

**Comment by masonwheeler** [comment]
@Priya91 That will help on your build machines, but it won't do much for those of us trying to build at home...

**Comment by Priya91** [comment]
Fixed the CI machines with image updates. The outerloop CI builds are green. Closing the issue

![image](https://cloud.githubusercontent.com/assets/7853870/13624736/94a39618-e565-11e5-8245-7a3d4dd4019e.png)
